### PR TITLE
refactor the r_print_raw function in libr/util/print.c to replace a s…

### PR DESCRIPTION
**Pull request by gogo2464**
…witch by a switch

**Checklist**

- [ ] Closing issues: #issue
- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

start to fix https://github.com/radareorg/radare2/issues/18632 . 

Not finished. Already fix a refactoring.

### CODE REVIEW

File: libr/util/print.c [hunk 1]
```diff
@@ -1485,56 +1485,64 @@ R_API void r_print_bytes(RPrint *p, const ut8 *buf, int len, const char *fmt) {
...
-					r_print_cursor (p, i + j, 1, 0);
+				p->cb_printf ("\n");
+			}
+			break;
+		}
+		default:
+		{
+			const ut8 *o, *q;
+			ut64 off;
+			int i, linenum_abs, mustbreak = 0, linenum = 1;
```
Comment (trufae) [inline libr/util/print.c:1519]: mustbreak should be `bool`, not int

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/util/print.c (hunk 1 above)
Comment (gogo2464) [reply libr/util/print.c:1519]: I just made a copy paste. I am going to solve.

**Comment by gogo2464** [comment]
Ready to merge. Just do some tests before and wait for the automatic tests. I have removed the check for the size. It just checks the length instead. I hope it does not break nothing.

File: libr/cons/cons.c [hunk 2]
```diff
@@ -1268,7 +1268,7 @@ R_API int r_cons_get_column(void) {
 
 /* final entrypoint for adding stuff in the buffer screen */
 R_API int r_cons_write(const char *str, int len) {
-	if (R_STR_ISEMPTY (str) || len < 0) {
+	if (len < 0) {
```
Comment (trufae) [inline libr/cons/cons.c:1271]: can you add ` r_return_val_if_fail (str && len <1);`

and make ` if (len == 0) return 0;`?

**Review by trufae — APPROVED** [review]
The PR LGTM! i have tested it and it seems to work as expected. Can you address the comment and add a test? i was thinking in having something like this:
```
wx 4400440044004400
pr 10 > $x
wff $x
x8
```

but seems like the $alias files are not working well with wff. so i'll merge and add a test after fixing that. thank you!

File: libr/cons/cons.c (hunk 2 above)
Comment (gogo2464) [reply libr/cons/cons.c:1271]: Yes I will do it this today.

Comment (trufae) [reply libr/cons/cons.c:1271]: I already did that i. Master. Do you want to extend alias files to support null bytes too? I can explain better about what are alias files and how to handle them in discord

**Pull request merged**